runtime/near-vm-runner/src/wasmer_runner.rs

```diff
@@ -342,15 +266,113 @@ pub(crate) fn run_wasmer0_module<'a>(
     (Some(logic.outcome()), err)
 }
 
-pub fn compile_wasmer0_module(code: &[u8]) -> bool {
-    wasmer_runtime::compile_with_config(
-        code,
-        wasmer_runtime::CompilerConfig { features: WASMER_FEATURES, ..Default::default() },
-    )
-    .is_ok()
-}
-
 pub(crate) fn wasmer0_vm_hash() -> u64 {
     // TODO: take into account compiler and engine used to compile the contract.
     42
 }
+
+pub(crate) struct Wasmer0Runtime;
+
+impl crate::runner::Runtime for Wasmer0Runtime {
+    fn run(
+        &self,
+        code: &ContractCode,
+        method_name: &str,
+        ext: &mut dyn External,
+        context: VMContext,
+        wasm_config: &VMConfig,
+        fees_config: &RuntimeFeesConfig,
+        promise_results: &[PromiseResult],
+        current_protocol_version: ProtocolVersion,
+        cache: Option<&dyn CompiledContractCache>,
+    ) -> (Option<VMOutcome>, Option<VMError>) {
+        let _span = tracing::debug_span!(target: "vm", "run_wasmer").entered();
```

Comment:
Curious: had we used `#[tracing::instrument]` here, would the name be just `run` for all versions of `run`, or would it be unambiguous?

Reply:
`#[tracing::instrument]` only uses the function name, so all versions would be `run`. `#[tracing::instrument]` wouldn't work great here for another reason as well – the arguments to this function don't really represent themselves very nicely as span fields, forcing to `skip` most of them....